src/DynamoCoreWpf/ViewModels/PackageManager/InstalledPackagesViewModel.cs

```diff
@@ -1,51 +1,257 @@
+        /// </summary>
+        /// <param name="dynamoViewModel">Back pointer to the dynamo view model</param>
+        /// <param name="model">Back pointer to the package loader</param>
         public InstalledPackagesViewModel(DynamoViewModel dynamoViewModel, PackageLoader model)
         {
             this.Model = model;
             this.dynamoViewModel = dynamoViewModel;
-            
+
             InitializeLocalPackages();
+            PopulateFilters();
+        }
+
+        /// <summary>
+        /// Filter the list of local packages based on the current filter selection
+        /// </summary>
+        internal void ApplyPackageFilter()
+        {
+            ClearPackagesViewModel();
+
+
+            if (NoActiveFilterSelected)
+            {
+                LocalPackages.AddRange(Model.LocalPackages.Select(NewPackageViewModel));
+            }
+            else
+            {
+                var currentFilter = CurrentFilterSelection;
+                foreach (var pkg in Model.LocalPackages)
+                {
+                    var viewModel = NewPackageViewModel(pkg);
+                    if (viewModel.PackageLoadStateText == currentFilter)
+                    {
+                        LocalPackages.Add(viewModel);
+                    }
+                    else
+                    {
+                        ClearPackageViewModel(viewModel);
+                    }
+                }
+            }
+            RaisePropertyChanged(nameof(LocalPackages));
         }
 
-        internal void GoToWebsite()
+        /// <summary>
+        /// Populate the list of available filters based on the packages currently known by the dynamo view model
+        /// </summary>
+        internal void PopulateFilters()
         {
-            dynamoViewModel.PackageManagerClientViewModel.GoToWebsite();
+            var currentSelection = CurrentFilterSelection;
+            Filters.Clear();
+            var filterNames = Model.LocalPackages.Select(pkg => new PackageViewModel(dynamoViewModel, pkg))
+                .Select(vm => vm.PackageLoadStateText).Distinct();
+            Filters.AddRange(filterNames.Select(f => new PackageFilter(f, this)));
+
+            if (Filters.Any())
+            {
+                Filters.Insert(0, new PackageFilter(this));
+            }
+
+            ResetCurrentSelection(currentSelection);
+
+            RaisePropertyChanged(nameof(Filters));
         }
 
-        private void InitializeLocalPackages()
+        private string CurrentFilterSelection => Filters.FirstOrDefault(f => f.IsChecked)?.Name;
+
+        private bool NoActiveFilterSelected
         {
-            foreach (var pkg in Model.LocalPackages)
+            get
             {
-                LocalPackages.Add(new PackageViewModel(this.dynamoViewModel, pkg));
+                var currentFilter = CurrentFilterSelection;
+                return currentFilter == null || currentFilter == Resources.PackageFilter_Name_All;
```

Comment:
I think you can just use `return CurrentFilterSelection== Resources.PackageFilter_Name_All` 
if CurrentFilterSelection is null then it is still a valid check (i.e null == Resources.PackageFilter_Name_All)

Reply:
Yes, but I want  the no filter (currentFilter is null) to be the same as a the All Filter.